lib/rest-adapter.js

```diff
@@ -346,8 +347,26 @@ RestAdapter.urlNotFoundHandler = function() {
   };
 };
 
+// singleton instance of strongErrorHandler
+RestAdapter.strongErrorHandler = null;
+
+// create or retrieve shared instance of strongErrorHandler
+RestAdapter.getStrongErrorHandler = function(options) {
```

Comment:
Sorry for the confusion. We should not use a true singleton here, because it's possible to create multiple `RestAdapter` instances, possibly configured with different `options`.

In my previous comments, I wanted you save the error handler instance in a local variable created inside `errorHandler` function but outside of `restErrorHandler` function.

Also I think `strongErrorHandler` was not meant to be called with `new`, since it returns another function, not an object.

```js
RestAdapter.errorHandler = function(options) {
  options = options || {};
  if (options.hasOwnProperty('disableStackTrace')) {
    // ...
  }

  var strongErrorHandlerInstance = strongErrorHandler(options);

  return function(restErrorHandler(err, req, res, next) {
    // ...
    strongErrorHandlerInstance(err, req, res, next);
  };
};
```